Composer/packages/client/src/pages/botProject/CreatePublishProfileDialog.tsx

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import React, { Fragment, useState, useEffect } from 'react';
+import { jsx } from '@emotion/core';
+import { useRecoilValue } from 'recoil';
+import { PublishTarget } from '@bfc/shared';
+import formatMessage from 'format-message';
+import { ActionButton, DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import { useBoolean } from '@uifabric/react-hooks';
+import Dialog, { DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
+
+import { dispatcherState, settingsState, publishTypesState } from '../../recoilModel';
+import { AuthDialog } from '../../components/Auth/AuthDialog';
+import { isShowAuthDialog } from '../../utils/auth';
+
+import { PublishProfileDialog } from './create-publish-profile/PublishProfileDialog';
+import { actionButton } from './styles';
+
+// -------------------- CreatePublishProfileDialog -------------------- //
+
+type CreatePublishProfileDialogProps = {
+  projectId: string;
+  scrollToSectionId?: string;
+};
+
+export const CreatePublishProfileDialog: React.FC<CreatePublishProfileDialogProps> = (props) => {
+  const { projectId } = props;
+  const { publishTargets } = useRecoilValue(settingsState(projectId));
+  const { getPublishTargetTypes, setPublishTargets } = useRecoilValue(dispatcherState);
+  const publishTypes = useRecoilValue(publishTypesState(projectId));
+
+  const [dialogHidden, setDialogHidden] = useState(true);
+  const [showAuthDialog, setShowAuthDialog] = useState(false);
+  const [hideDialog, { toggle: toggleHideDialog }] = useBoolean(false);
+
+  const dialogTitle = {
```

Comment:
nit: move the formatMessage to where they are used, it's not being reused anywhere else and it's not being memoized so no need to have this object.

Reply:
the dialogTitle is used in render element, so i use formatMessage to wrap it. and title & subTitle is combined as an object for `<Dialog />` component